src/main/cpp/t04_students.cpp

```diff
@@ -29,10 +29,100 @@
 
 #include "t04_students.h"
 #include <iostream>
-
+#include <map>
+#include <cmath>
+#include <string>
+#include <algorithm>
 using namespace std;
 
 int t04_students() {
-    
+	int n,p=0;
+	double order1[23];
+	multimap<double, string> m;
+	string ah,order2[20];
+	cin >> n;
+	getline(cin, ah);
+	for (int i = 0; i < n; i++)
+	{
+		bool snums = false;
+		string s2 = "",h="";
+		string  name = "", nums = "";
+		long long g = -1,ngh=0;
+		while (ngh!=3)
+		{
+			if ((int)ah[p] > 48 && (int)ah[p] < 54)
+				ngh++;
+			s2 += ah[p];
+			p++;
+		}
+		h = s2;
+		do
+		{
+			g++;
+			if ((int)h[g] > 48 && (int)h[g] < 54)
+			{
+				snums = true;
+			}
+			if (snums)
+			{
+				nums += h[g];
+			}
+			else if (((int )h[g + 1] > 48 && (int)h[g + 1] < 54 && snums == false )&& (g+1)<h.length())
+				continue;
+			else if (snums == false)
+			{
+				name += h[g];
+			}
+		} while (h[g] != '\0');
+		char h1 = nums[0], h2 = nums[2], h3 = nums[4];
+		double a1 = h1 - '0';
+		double a2 = h2 - '0';
+		double a3 = h3 - '0';
+		order2[i] = name;
+		order1[i] = (a1 + a2 + a3) / 3;
+		m.insert(make_pair((a1 + a2 + a3) / 3, name));
+	}
+	int h = 0;
+	double a1 = 0, a2 = 0;
+	for (multimap<double, string>::iterator it = m.end(); it != m.begin();)
+	{
+		a1 = 0;
+		a2 = 0;
+		--it;
+		if (h <= n - 2)
+		{
+			a1 = (--it)->first;
+			it++;
+		}
+		if (h != 0)
+		{
+			a2 = ((++it)->first);
+			it--;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Prefer prefix ++/-- operators for non-primitive types.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4586360)